```
[ 94%] Generating avrdude.info
Undefined subroutine &Locale::Messages::dgettext called at /nix/store/n80nqbxmqnx0k3x0kx2zj87s4xy1ia1q-texi2html-5.0/bin/texi2html line 29628.
make[2]: *** [src/doc/CMakeFiles/html.dir/build.make:78: src/doc/avrdude-html/avrdude.html] Error 2
make[1]: *** [CMakeFiles/Makefile2:343: src/doc/CMakeFiles/html.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 94%] Built target info
[ 96%] Built target dvi
[ 98%] Built target pdf
[ 98%] Built target ps
make: *** [Makefile:136: all] Error 2
```

--- a/pkgs/development/embedded/avrdude/default.nix
+++ b/pkgs/development/embedded/avrdude/default.nix
@@ -1,9 +1,7 @@
-{ lib, stdenv, fetchFromGitHub, cmake, bison, flex, libusb-compat-0_1, libelf, libftdi1, readline
-# docSupport is a big dependency, disabled by default
-, docSupport ? false, texLive ? null, texinfo ? null, texi2html ? null
-}:
-
-assert docSupport -> texLive != null && texinfo != null && texi2html != null;
+{ lib, stdenv, fetchFromGitHub, cmake, bison, flex, libusb-compat-0_1, libelf
+, libftdi1, readline
+# documentation building is broken on darwin
+, docSupport ? (!stdenv.isDarwin), texlive, texinfo, texi2html, unixtools }:
 
 stdenv.mkDerivation rec {
   pname = "avrdude";
@@ -16,10 +14,14 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-pGjOefWnf11kG/zFGwYGet1OjAhKsULNGgh6vqvIQ7c=";
   };
 
-  nativeBuildInputs = [ cmake bison flex ];
+  nativeBuildInputs = [ cmake bison flex ] ++ lib.optionals docSupport [
+    unixtools.more
+    texlive.combined.scheme-medium
+    texinfo
+    texi2html
+  ];
 
-  buildInputs = [ libusb-compat-0_1 libelf libftdi1 readline ]
-    ++ lib.optionals docSupport [ texLive texinfo texi2html ];
+  buildInputs = [ libusb-compat-0_1 libelf libftdi1 readline ];
 
   cmakeFlags = lib.optionals docSupport [
     "-DBUILD_DOC=ON"

